src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -1267,32 +1267,39 @@ trait Contexts { self: Analyzer =>
           cx = cx.outer // push further outward
       }
 
-      val defPre = pre
-      val defCx = cx
-
-      while (foundInSuper && (cx ne NoContext) && (cx.scope ne null)) {
-        pre = cx.enclClass.prefix
-        val next = lookupInScope(cx.owner, cx.enclClass.prefix, cx.scope).orElse(searchPrefix)
-        if (next.exists && next.owner == cx.owner && thisContext.unit.exists && next.sourceFile == thisContext.unit.source.file) {
-          outerDefSym = next
-          cx = NoContext
-        } else
-          cx = cx.outer
-      }
-      if (outerDefSym.exists) {
-        if ((defSym.isAliasType || outerDefSym.isAliasType) && defPre.memberType(defSym) =:= pre.memberType(outerDefSym))
-          outerDefSym = NoSymbol
-        if (defSym.isStable && outerDefSym.isStable &&
-          (pre.memberType(outerDefSym).termSymbol == defSym || defPre.memberType(defSym).termSymbol == outerDefSym))
-          outerDefSym = NoSymbol
-      }
-
-      pre = defPre
-      cx = defCx
-
       if (symbolDepth < 0)
         symbolDepth = cx.depth
 
+      def checkAmbiguousWithEnclosing(): Unit = if (foundInSuper && !thisContext.unit.isJava) {
```

Comment:
`isJava` is added; in 2.13: https://github.com/scala/scala/blob/c47f6632b97432f5745a9f02d1bc067709d9d6a1/src/compiler/scala/tools/nsc/typechecker/Contexts.scala#L1598